### PR TITLE
improve Spherical docs

### PR DESCRIPTION
**Pull request by jw3126**

### CODE REVIEW

**Comment by andyferris** [comment]
Thanks. We may want to switch conventions in the future - but it is good to at least document what is implemented :)

Is this good to go?

**Review by andyferris — APPROVED** [review]

**Comment by jw3126** [comment]
I think it is good to go.

File: src/coordinatesystems.jl [hunk 1]
```diff
@@ -73,6 +73,26 @@ Base.convert(::Type{Polar}, v::AbstractVector) = PolarFromCartesian()(v)
 #############################
 """
 Spherical(r, θ, ϕ) - 3D spherical coordinates
+
+This uses the so called mathematics convention as opposed to the physics convention.
```
Comment (waldyrious) [inline src/coordinatesystems.jl:77]:
```suggestion
This uses the so-called mathematics convention as opposed to the physics convention.
```

File: src/coordinatesystems.jl [hunk 2]
```diff
@@ -73,6 +73,26 @@ Base.convert(::Type{Polar}, v::AbstractVector) = PolarFromCartesian()(v)
...
+Given the cartesian coordinates `x,y,z` of a vector `v`, the polar coordinates are:
```
Comment (waldyrious) [inline src/coordinatesystems.jl:78]:
```suggestion
Given the Cartesian coordinates `x,y,z` of a vector `v`, the polar coordinates are:
```

File: src/coordinatesystems.jl [hunk 3]
```diff
@@ -73,6 +73,26 @@ Base.convert(::Type{Polar}, v::AbstractVector) = PolarFromCartesian()(v)
...
+
+* `r` is the radius. It is given by `norm(v, 2)`.
+* `θ` is the azimuthal angle. It is the angle from the x-axis to the project of `v` onto the xy-plane.
```
Comment (waldyrious) [inline src/coordinatesystems.jl:81]:
```suggestion
* `θ` is the azimuthal angle. It is the angle from the x-axis to the projection of `v` onto the xy-plane.
```

**Comment by jw3126** [comment]
Merge?

**Review by c42f — COMMENTED** [review]
I'm not sure this is quite right, did you read through #25?

File: src/coordinatesystems.jl [hunk 4]
```diff
@@ -73,6 +73,26 @@ Base.convert(::Type{Polar}, v::AbstractVector) = PolarFromCartesian()(v)
 #############################
 """
 Spherical(r, θ, ϕ) - 3D spherical coordinates
+
+This uses the so-called mathematics convention as opposed to the physics convention.
+Given the Cartesian coordinates `x,y,z` of a vector `v`, the polar coordinates are:
+
+* `r` is the radius. It is given by `norm(v, 2)`.
+* `θ` is the azimuthal angle. It is the angle from the x-axis to the projection of `v` onto the xy-plane.
+* `ϕ` is the polar angle. It is the angle from the z-axis to `v`.
```
Comment (c42f) [inline src/coordinatesystems.jl:82]: In particular, IIRC we had a somewhat dodgy/weird definition and `ϕ` was the latitude rather than the polar angle.

Comment (jw3126) [reply src/coordinatesystems.jl:82]: Dang you are right!

File: src/coordinatesystems.jl [hunk 5]
```diff
@@ -73,6 +73,25 @@ Base.convert(::Type{Polar}, v::AbstractVector) = PolarFromCartesian()(v)
 #############################
 """
 Spherical(r, θ, ϕ) - 3D spherical coordinates
+
+There are many Spherical coordinate conventions and this library uses a somewhat exotic one.
+Given a vector `v` with cartesian coordinates `xyz`, let `v_xy = [x,y,0]` be the orthogonal projection on the `xy` plane.
```
Comment (waldyrious) [inline src/coordinatesystems.jl:78]: Sorry to be repetitive 😅

```suggestion
Given a vector `v` with Cartesian coordinates `xyz`, let `v_xy = [x,y,0]` be the orthogonal projection of `v` on the `xy` plane.
```

File: test/runtests.jl [hunk 6]
```diff
@@ -4,9 +4,12 @@ using CoordinateTransformations
 using ForwardDiff: Dual, partials
 using StaticArrays
 using Unitful
+using Documenter: doctest
+
 
 @testset "CoordinateTransformations" begin
 
+    doctest(CoordinateTransformations)
```
Comment (c42f) [inline test/runtests.jl:12]: Cool, I didn't know about this feature of `Documenter` :+1:

**Comment by c42f** [comment]
Thanks for this. By the way, if you wanted to correct the situation to have a more normal definition of this angle, overriding `getproperty` with some more explicit names would likely help a lot.

See also #25 for other options for improving the situation.

Regardless, thanks for the PR!

**Pull request merged**